docs/docs/dynamo-store/architecture.md

```diff
@@ -0,0 +1,117 @@
+
+<h2 id="event-schema">Event schema</h2>
+
+| Key             | Type    | Description                                                  |
+| --------------- | ------- | ------------------------------------------------------------ |
+| `t`imestamp     | String  | ISO8601 DateTime representing the time the event was written |
+| `D`ata encoding | ?Number | the encoding used for the `d`ata attribute                   |
+| `d`ata          | ?Binary | The event payload                                            |
+| `M`eta encoding | ?Number | the encoding used for the `m`eta attribute                   |
+| `m`eta          | ?Binary | The event metadata                                           |
+| `x`actionId     | ?String | The correlation id of the event                              |
+| wh`y`           | ?String | The causation id (wh`y` did this event occur)                |
+
+<h2 id="unfold-schema">Unfold schema</h2>
+
+| Key             | Type    | Description                                                  |
+| --------------- | ------- | ------------------------------------------------------------ |
+| `i`ndex         | Number  | The index at which the unfold was generated                  |
+| `c`ase          | String  | The event type of the unfold                                 |
+| `t`imestamp     | String  | ISO8601 DateTime representing the time the event was written |
+| `D`ata encoding | ?Number | the encoding used for the `d`ata attribute                   |
+| `d`ata          | ?Binary | The event payload                                            |
+| `M`eta encoding | ?Number | the encoding used for the `m`eta attribute                   |
+| `m`eta          | ?Binary | The event metadata                                           |
+
+# Reactions
+
+Reactions are an often necessary part of event-sourced architectures. DynamoDB
+offers DynamoDB Streams, a Change Data Capture (CDC) for your table. This CDC
+feed offers two guarantees
+
+- Each stream record appears exactly once in the stream
+- For each item that is modified in a DynamoDB table, the stream records appear
+  in the same sequence as the actual modifications to the item.
+- The stream contains the last 24 hours of updates
+
+You'll notice that the second guarantee notably does not guarantee ordering
+across items within the same hash key, only for updates to a singular (hash,
+range) pair. For this reason every event written to DynamoStore will be first
+written to the Tip document.
+
+The third guarantee means DynamoDB streams cannot be used for general purpose
+crawling of all events in the store. One of the main selling points of
+event-sourced architectures is that you are able to create a new read-model as
+if it had always been there by processing all events in the store. To enable
+this we construct an index. This index lives in a secondary table and is kept up
```

Comment:
I know secondary alludes to secondary index etc, but its a thing of its own. 'sister', 'associated', 'separate related' ?